paddle/fluid/inference/tests/api/tester_helper.h

```diff
@@ -463,68 +465,90 @@ void SummarizeAccuracy(float avg_acc_fp32, float avg_acc_int8,
             << std::setprecision(4) << avg_acc_int8;
 }
 
+void SummarizePerformance(const char *title, float sample) {
+  CHECK_GT(sample, 0.0);
+  auto throughput = 1000.0 / sample;
+  LOG(INFO) << title << ": avg fps: " << std::fixed << std::setw(6)
+            << std::setprecision(4) << throughput << ", avg latency: " << sample
+            << " ms";
+}
+
 void SummarizePerformance(float sample_latency_fp32,
                           float sample_latency_int8) {
-  // sample latency in ms
-  auto throughput_fp32 = 1000.0 / sample_latency_fp32;
-  auto throughput_int8 = 1000.0 / sample_latency_int8;
-  LOG(INFO) << "--- Performance summary --- ";
-  LOG(INFO) << "FP32: avg fps: " << std::fixed << std::setw(6)
-            << std::setprecision(4) << throughput_fp32
-            << ", avg latency: " << sample_latency_fp32 << " ms";
-  LOG(INFO) << "INT8: avg fps: " << std::fixed << std::setw(6)
-            << std::setprecision(4) << throughput_int8
-            << ", avg latency: " << sample_latency_int8 << " ms";
+  if (FLAGS_enable_fp32) SummarizePerformance("FP32", sample_latency_fp32);
+  if (FLAGS_enable_int8) SummarizePerformance("INT8", sample_latency_int8);
 }
 
-void CompareAccuracy(
-    const std::vector<std::vector<PaddleTensor>> &output_slots_quant,
-    const std::vector<std::vector<PaddleTensor>> &output_slots_ref,
+float CompareAccuracyOne(
+    const std::vector<std::vector<PaddleTensor>> &output_slots,
     int compared_idx) {
-  if (output_slots_quant.size() == 0 || output_slots_ref.size() == 0)
+  if (output_slots.size() == 0)
     throw std::invalid_argument(
         "CompareAccuracy: output_slots vector is empty.");
 
-  float total_accs_quant{0};
-  float total_accs_ref{0};
-  for (size_t i = 0; i < output_slots_quant.size(); ++i) {
-    if (compared_idx == 1) {
-      PADDLE_ENFORCE_GE(
-          output_slots_quant[i].size(), 2UL,
-          "To achieve top 1 accuracy, output_slots_quant[i].size()>=2");
-      PADDLE_ENFORCE_GE(
-          output_slots_ref[i].size(), 2UL,
-          "To achieve top 1 accuracy, output_slots_ref[i].size()>=2");
-    } else if (compared_idx == 2) {
-      PADDLE_ENFORCE_GE(output_slots_quant[i].size(), 3UL,
-                        "To achieve mAP, output_slots_quant[i].size()>=3");
-      PADDLE_ENFORCE_GE(output_slots_ref[i].size(), 3UL,
-                        "To achieve mAP, output_slots_ref[i].size()>=3");
-    } else {
-      throw std::invalid_argument(
-          "CompareAccuracy: compared_idx is out of range.");
+  float total_accs{0};
+
+  for (size_t i = 0; i < output_slots.size(); ++i) {
+    switch (compared_idx) {
+      case 1:
+        PADDLE_ENFORCE_GE(
+            output_slots[i].size(), 2UL,
+            "To achieve top 1 accuracy, output_slots_quant[i].size()>=2");
+        break;
+
```

Comment:
Could you please remove the following while-line?

Reply:
done